src/group_theory/free_abelian_group.lean

```diff
@@ -345,7 +398,72 @@ instance [comm_monoid α] : comm_ring (free_abelian_group α) :=
       { intros y1 y2 ih1 ih2, rw [mul_add, add_mul, ih1, ih2] } },
     { intros s ih, rw [neg_mul_eq_neg_mul_symm, ih, neg_mul_eq_mul_neg] },
     { intros x1 x2 ih1 ih2, rw [add_mul, mul_add, ih1, ih2] }
-  end
+  end,
   .. free_abelian_group.ring α }
 
+instance pempty_unique : unique (free_abelian_group pempty) :=
+{ default := 0,
+  uniq := λ x, free_abelian_group.induction_on x rfl
+    (λ x, pempty.elim x)
+    (λ x, pempty.elim x)
+    (by { rintros - - rfl rfl, simp })  }
+
+
+/-- The free abelian group on the empty type is the trivial group. -/
+def pempty_equiv : free_abelian_group pempty ≃+ unit :=
+{ map_add' := λ _ _, unit.ext,
+  ..equiv_punit_of_unique }
```

Comment:
This probably belongs as a def in `equiv/mul_add`, `add_equiv_punit_of_unique`

Reply:
Or perhaps more generally, all of these should exist:
```
example {A B : Type*} [unique A] [unique B] : A ≃ B := equiv_of_unique_of_unique

example {A B : Type*} [unique A] [unique B] [add_monoid A] [add_monoid B] : A ≃+ B := by library_search
example {A B : Type*} [unique A] [unique B] [monoid A] [monoid B] : A ≃* B := by library_search
example {A B : Type*} [unique A] [unique B] [semiring A] [semiring B] : A ≃+* B := by library_search
```
and then this equiv might not be needed at all.

Reply:
I'll do this in another PR.

Reply:
#6079